### ISSUE TITLE
How to save detections which are counted as False Positives

### ISSUE THREAD

**Issue by ronit450** [ultralytics/ultralytics#19259]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

So I am doing yolo multiband and dealing with 7 band data. The issue I am having is there are alot of FPs in my detection and I want to know at what point or what is causing the surge in FPs. Can I somehow store those and test it?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ronit450, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset details, image examples, training logs, and metrics observed. Additionally, you can explore our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to refine your training processes.

For your question about false positives, YOLO’s predictions can be analyzed by saving the results of incorrect detections. Depending on your use case, this might involve working with the validation predictions or customizing the code to extract the false positives after inference. Should you need further assistance in implementation, please provide additional details such as the version of Ultralytics YOLO you are using and a snippet of your workflow.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

📢 Please note that this is an automated response. An Ultralytics engineer will assist you further as soon as possible.

**Comment by Y-T-G** [comment]
If you have a validation set, you can use this.

https://github.com/ultralytics/ultralytics/pull/18868

**Comment by ronit450** [comment]
No I want it to Save detections after each epoch

**Comment by Y-T-G** [comment]
I don't know what that means

**Comment by ronit450** [comment]
So I want to save detection at the end of each epoch to visualize how well the model is learning.

**Comment by Y-T-G** [comment]
You can probably modify the code slightly to do that.

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has been merged in PR #18868, which introduces enhanced validation visualizations for Detect, Segment, Pose, and OBB tasks. With the new visualize option, users can now see ground truths, true positives, false positives, and false negatives per image—making debugging and understanding model performance much more intuitive. The update also improves internal handling of confusion matrices, updates related tests, and refreshes the documentation to help users get started with the new feature.

Special thanks to @Y-T-G for authoring the PR, and to @Laughing-q, @Skillnoob, @glenn-jocher, @Burhan-Q, @Petros626, and @O-J1 for their valuable contributions!

To test if these changes resolve the issue, try updating to the latest version with:
```
pip install -U ultralytics>=8.3.169
```
Alternatively, you can install directly from the main branch:
```
pip install git+https://github.com/ultralytics/ultralytics.git@main
```
Or, if you prefer, you can wait for the next official PyPI release.

Please let us know if this update addresses your issue! Thank you for reporting this, and feel free to share additional feedback if the problem persists. 🙏
